### PR TITLE
Fixed python 3 incompatibility in logging API implemenation of findCaller

### LINKED ISSUE #5057: ST2 logs writing "(unknown file)" in log prefix, instead of module name
## SUMMARY

In ST2 v3.2, the st2sensorcontainer.log, and in v3.3 all st2 logs have the `module` name replaced with `(unknown file)`

### STACKSTORM VERSION

v3.2, v3.3, EL8 and U18.04
Python 3.6

### OS, environment, install method

One Line and Ansible so far. 

## Steps to reproduce the problem

Show how to reproduce the problem, using a minimal test-case. Make sure to include any content
(pack content - workflows, actions, etc.) which are needed to reproduce the problem.

## Expected Results

No `(unknown file)`, but the python module name. 

## Actual Results
```
2020-10-13 17:55:13,787 140460262501416 INFO (unknown file) [-] Sensor linux.FileWatchSensor started
2020-10-13 17:55:15,444 139725927337456 INFO (unknown file) [-] No config found for sensor "FileWatchSensor"
2020-10-13 17:55:15,446 139725927337456 INFO (unknown file) [-] Watcher started
2020-10-13 17:55:15,446 139725927337456 INFO (unknown file) [-] Running sensor initialization code
2020-10-13 17:55:15,454 139725927337456 INFO (unknown file) [-] Running sensor in passive mode
```

### PR DESCRIPTION
**Pull request by nmaludy**
Closes #5057 

See that issue for description of what the problem was. The fix was simply to ensure our implementation handles both the python 2 version and returns a 3-tuple, along with the python 3 version which returns a 4-tuple and is based on the code here: https://github.com/python/cpython/blob/3.9/Lib/logging/__init__.py#L1502-L1536

I've tested this change both my a python 2.7 (centos7) and python 3.6 (centos8 and ubuntu18) box and the logging modules are returned properly.

### CODE REVIEW

**Review by m4dcoder — APPROVED** [review]

File: st2common/st2common/log.py [hunk 1]
```diff
@@ -69,25 +70,64 @@
...
     Note: This is based on logging/__init__.py:findCaller and modified so it takes into account
-    this file - https://hg.python.org/cpython/file/2.7/Lib/logging/__init__.py#l1233
+    this file:
+    https://github.com/python/cpython/blob/2.7/Lib/logging/__init__.py#L1240-L1259
+
+    The Python 3.x implementation adds in a new argument `stack_info` and `stacklevel`
+    and expects a 4-element tuple to be returned, rather than a 3-element tuple in
+    the python 2 implementation.
+    We derived our implementation from the Python 3.9 source code here:
+    https://github.com/python/cpython/blob/3.9/Lib/logging/__init__.py#L1376-L1404
```
Comment (cognifloyd) [inline st2common/st2common/log.py:86]:
```suggestion
    https://github.com/python/cpython/blob/3.9/Lib/logging/__init__.py#L1502-L1536
```

**Review by cognifloyd — CHANGES_REQUESTED** [review]
From reading and comparing with the python code, LGTM.
Though it seems there have been changes in 3.9 since you looked, so the line numbers have changed.

**Comment by m4dcoder** [comment]
@nmaludy Looks like there's an issue starting sensor container. The integration test in travis is failing consistently on both PRs.

**Comment by nmaludy** [comment]
@cognifloyd good catch, i had the right line numbers in the description of this PR, but must have had a bad copy/paste in the code itself.

@m4dcoder i re-ran the integration tests and they passed (i just committed @cognifloyd  change so we'll see how they run this time), also note that i tested sensors explicitly since that is what @punkrokk reported and they definitely worked on my systems (centos7, centos8, ubuntu18)

**Review by cognifloyd — APPROVED** [review]
LGTM

**Comment by m4dcoder** [comment]
@nmaludy @cognifloyd LGTM Thanks guys!

**Pull request merged**